### PR TITLE
Add section about Sphinx required configuration 

### LINKED ISSUE #150: Add recommended configuration of Sphinx build options
### 📝 Description of the feature

@dnwillia-work thought it would be good to have the recommended configuration for Sphinx build options in the *PyAnsys Developer's Guide*. He said that Fluent's build options are here: [pyfluent/Makefile at main · pyansys/pyfluent (github.com)](https://github.com/pyansys/pyfluent/blob/main/doc/Makefile)
He then said that they use ``-W`` and that most repos should probably be using that.

### 💡 Steps for implementing the feature

_No response_

### 🔗 Useful links and references

Fluent's Sphinx build options are here: [pyfluent/Makefile at main · pyansys/pyfluent (github.com)](https://github.com/pyansys/pyfluent/blob/main/doc/Makefile)

### PR DESCRIPTION
**Pull request by jorgepiloto**
Resolves #150 and includes example `conf.py`, `Makefile` and `make.bat` files.

### CODE REVIEW

File: doc/source/doc-style/code/required_make.bat [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+set BUILDDIR=build
+set SPHINXOPTS=-j auto -W --keep-going
+
+if "%1" == "" goto help
+if "%1" == "pdf" goto pdf
+
+%SPHINXBUILD% >NUL 2>NUL
+if errorlevel 9009 (
+	echo.
+	echo.The 'sphinx-build' command was not found. Make sure you have Sphinx
```
Comment (PipKat) [inline doc/source/doc-style/code/required_make.bat:20]: @jorgepiloto Is it OK for me to edit these "echo" lines?

Comment (jorgepiloto) [reply doc/source/doc-style/code/required_make.bat:20]: Sure, no problem doing that @PipKat.

File: doc/source/doc-style/doc-configuration.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
+Required Sphinx configuration
+=============================
+The following page explains the minimum required `Sphinx`_ configuration for
+building the documentation of a PyAnsys project.
+
+When installing `Sphinx`_, a program named ``sphinx-build`` gets intalled too.
```
Comment (MaxJPRey) [inline doc/source/doc-style/doc-configuration.rst:6]:
```suggestion
When installing `Sphinx`_, a program named ``sphinx-build`` gets installed too.
```

File: doc/source/doc-style/doc-configuration.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+This program is in chager of collecting, parsing and rendering all the
```
Comment (MaxJPRey) [inline doc/source/doc-style/doc-configuration.rst:7]:
```suggestion
This program is in charge of collecting, parsing and rendering all the
```

File: doc/source/doc-style/doc-configuration.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ReStructured Text files in :ref:`The \`\`doc/\`\` directory`.
+
+The behaviour of ``sphinx-build`` is controlled through a ``Makefile`` (to be
```
Comment (MaxJPRey) [inline doc/source/doc-style/doc-configuration.rst:10]:
```suggestion
The behavior of ``sphinx-build`` is controlled through a ``Makefile`` (to be
```

File: doc/source/doc-style/doc-configuration.rst [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+``conf.py`` file. Custom additions should be done on top of this configuration
+to guarantee that a minimum consistency applies across different PyAnsys
+projects.
+
+.. literalinclude:: code/required_conf.py
+
+Automation files
+----------------
+As introduced at the beginning of this section, the ``sphinx-build`` program and
+all its options and arguments can be automated trhought the usage of a
```
Comment (MaxJPRey) [inline doc/source/doc-style/doc-configuration.rst:28]:
```suggestion
all its options and arguments can be automated trough the usage of a
```

File: doc/source/doc-style/code/required_makefile [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# Minimal makefile for Sphinx documentation
```
Comment (MaxJPRey) [inline doc/source/doc-style/code/required_makefile:1]: @jorgepiloto Really useful!

**Review by MaxJPRey — APPROVED** [review]

File: doc/source/doc-style/code/required_conf.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    "GL09",  # Deprecation warning should precede extended summary
+    "GL10",  # reST directives {directives} must be followed by two colons
+    "SS01",  # No summary found
+    "SS02",  # Summary does not start with a capital letter
+    # "SS03", # Summary does not end with a period
+    "SS04",  # Summary contains heading whitespaces
+    # "SS05", # Summary must start with infinitive verb, not third person
+    "RT02",  # The first line of the Returns section should contain only the
+    # type, unless multiple values are being returned"
+}
```
Comment (RobPasMue) [inline doc/source/doc-style/code/required_conf.py:69]:
```suggestion
    "SS02",  # Summary does not start with a capital letter
    "SS03", # Summary does not end with a period
    "SS04",  # Summary contains heading whitespaces
    "SS05", # Summary must start with infinitive verb, not third person
    "RT02",  # The first line of the Returns section should contain only the
    # type, unless multiple values are being returned"
}
```

I would enforce all rules by default - not leave them commented out. However, if you feel that they are not being applied anywhere and we should just delete them, please go ahead =)

File: doc/source/doc-style/code/required_make.bat [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+:help
+%SPHINXBUILD% -M help %SOURCEDIR% %BUILDDIR% %SPHINXOPTS% %O%
+
+:pdf
+	%SPHINXBUILD% -M latex %SOURCEDIR% %BUILDDIR% %SPHINXOPTS% %O%
+	cd "%BUILDDIR%\latex"
+	for %%f in (*.tex) do (
+	pdflatex "%%f" --interaction=nonstopmode)
+
```
Comment (RobPasMue) [inline doc/source/doc-style/code/required_make.bat:41]:
```suggestion
if "%1" == "" goto help
if "%1" == "clean" goto clean
if "%1" == "pdf" goto pdf

%SPHINXBUILD% >NUL 2>NUL
if errorlevel 9009 (
	echo.
	echo.The 'sphinx-build' command was not found. Make sure you have Sphinx
	echo.installed, then set the SPHINXBUILD environment variable to point
	echo.to the full path of the 'sphinx-build' executable. Alternatively you
	echo.may add the Sphinx directory to PATH.
	echo.
	echo.If you don't have Sphinx installed, grab it from
	echo.http://sphinx-doc.org/
	exit /b 1
)

%SPHINXBUILD% -M %1 %SOURCEDIR% %BUILDDIR% %SPHINXOPTS% %O%
goto end

:help
%SPHINXBUILD% -M help %SOURCEDIR% %BUILDDIR% %SPHINXOPTS% %O%

:clean
rmdir /s /q %BUILDDIR% > /NUL 2>&1 
for /d /r %SOURCEDIR% %%d in (_autosummary) do @if exist "%%d" rmdir /s /q "%%d"
goto end

:pdf
	%SPHINXBUILD% -M latex %SOURCEDIR% %BUILDDIR% %SPHINXOPTS% %O%
	cd "%BUILDDIR%\latex"
	for %%f in (*.tex) do (
	pdflatex "%%f" --interaction=nonstopmode)
	
```

Let's add the clean option as well

File: doc/source/doc-style/doc-configuration.rst [hunk 9]
```diff
@@ -0,0 +1,68 @@
+<!-- vale off -->
+Required Sphinx configuration
+=============================
+<!-- vale on -->
```
Comment (RobPasMue) [inline doc/source/doc-style/doc-configuration.rst:4]: Why are you using vale off/on ? I'd prefer that you add "Sphinx" to the doc/styles/Vocab/accept.txt file. That is the place where "ignored" or "admitted" words are located

**Review by RobPasMue — APPROVED** [review]
LGTM apart from my comments =) Thanks @jorgepiloto

File: doc/source/doc-style/code/required_conf.py (hunk 7 above)
Comment (jorgepiloto) [reply doc/source/doc-style/code/required_conf.py:69]: Sure, no problem!

File: doc/source/doc-style/code/required_make.bat (hunk 8 above)
Comment (jorgepiloto) [reply doc/source/doc-style/code/required_make.bat:41]: Good catch!

File: doc/source/doc-style/doc-configuration.rst (hunk 9 above)
Comment (jorgepiloto) [reply doc/source/doc-style/doc-configuration.rst:4]: Oh! I tried to add it to the `accept.txt` but Vale then complained about `sphinx` without capital case. I think if I add both it should work, let me check.

Comment (RobPasMue) [reply doc/source/doc-style/doc-configuration.rst]: Yep, that should work :)

**Pull request merged**